builtin/providers/azurerm/resource_arm_storage_account.go

```diff
@@ -131,16 +132,47 @@ func resourceArmStorageAccountCreate(d *schema.ResourceData, meta interface{}) e
 		Tags:     expandTags(tags),
 	}
 
-	_, err := storageClient.Create(resourceGroupName, storageAccountName, opts, make(chan struct{}))
-	if err != nil {
-		return fmt.Errorf("Error creating Azure Storage Account '%s': %s", storageAccountName, err)
+	// Create the storage account. We wrap this so that it is cancellable
+	// with a Ctrl-C since this can take a LONG time.
+	wrap := signalwrapper.Run(func(cancelCh <-chan struct{}) error {
+		_, err := storageClient.Create(resourceGroupName, storageAccountName, opts, cancelCh)
+		return err
+	})
+
+	// Check the result of the wrapped function. I put this into a select
+	// since we will likely also want to introduce a time-based timeout.
+	var createErr error
+	select {
+	case <-time.After(1 * time.Hour):
+		// An hour is way above the expected P99 for this API call so
```

Comment:
We may get to see this in action during nightly acceptance tests if they have any issues - that's when we tend to get a lot of failures